src/concurrency/thread.rs

```diff
@@ -656,11 +648,18 @@ impl<'tcx> ThreadManager<'tcx> {
                     @capture<'tcx> {
                         joined_thread_id: ThreadId,
                     }
-                    @unblock = |this| {
-                        if let Some(data_race) = &mut this.machine.data_race {
-                            data_race.thread_joined(&this.machine.threads, joined_thread_id);
+                    @unblock = |this, tcb_state| {
+                        match tcb_state {
```

Comment:
```suggestion
                    @unblock = |this, unblock| {
```
Please also use this name for all the other unblock callbacks.